rq/worker.py

```diff
@@ -711,6 +712,7 @@ def perform_job(self, job, queue):
                 started_job_registry=started_job_registry
             )
         except Exception:
+            job.refresh()
```

Comment:
Could you please move `job.refresh()` to just before `job.save()` inside `handle_job_failure()`?